src/vm/compiler.rs

```diff
@@ -134,15 +137,30 @@ impl Compiler {
                         //
                         // This value is the field's default value.
                         // Otherwise its initalized to `nil`.
-                        match &field_assign.value {
+                        match &assign.value {
                             Some(value) => self.compile_expr(value, gc)?,
-                            None => self.emit_u8(op::NIL, span),
+                            None => self.emit_u8(op::NIL, &span),
                         }
 
-                        self.emit_u8(op::FIELD, span);
+                        self.emit_u8(op::FIELD, &span);
                         // Emit field name's constant index
                         let name = gc.alloc(name).into();
-                        self.emit_constant(name, span)?;
+                        self.emit_constant(name, &span)?;
+
+                        let access_modifier = match &field_assign.access_modifier {
+                            // Emit field's access modifier
+                            Some(x) => x,
+                            None => &AccessModifier::Public,
+                        };
+
+                        let access_modifier_string = match access_modifier {
+                            AccessModifier::Invalid => String::from("invalid"),
+                            AccessModifier::Private => String::from("private"),
+                            AccessModifier::Public => String::from("public"),
+                        };
+
+                        let access_modifier_idx = gc.alloc(&access_modifier_string).into();
+                        self.emit_constant(access_modifier_idx, span)?;
```

Comment:
Emit byte op with the index of the access modifier string in the constants table. This implementation will likely change to using bit flags: https://docs.rs/bitvec/1.0.1/bitvec/slice/struct.BitSlice.html#method.from_element